dsc/s3.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
+
+from boto3 import client
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+
+    from mypy_boto3_s3.type_defs import PutObjectOutputTypeDef
+
+logger = logging.getLogger(__name__)
+
+
+class S3Client:
+    """An S3 class that provides a generic boto3 s3 client."""
```

Comment:
I might propose a docstring update here?  If it truly is just providing a generic s3 client, then it probably isn't of much use!  

I'd suggest something along the lines of,
> "Client to perform common S3 operations for this application."

It's subtle, and maybe nit-picky, but I think it hints at the methods defined are aligned with the goals of the app.  It's _not_ very opinionated, it is quite generic, but that at least explains why these methods (i.e. they are things this app needs to do).